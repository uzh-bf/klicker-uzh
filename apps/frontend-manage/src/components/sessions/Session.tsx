import { useMutation } from '@apollo/client'
import { faCalendarDays, faPlay } from '@fortawesome/free-solid-svg-icons'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import {
  QuestionInstance,
  Session as SessionType,
  SessionBlock,
  StartSessionDocument,
} from '@klicker-uzh/graphql/dist/ops'
import { Button, H2, H4 } from '@uzh-bf/design-system'
import dayjs from 'dayjs'
import Link from 'next/link'
import { useRouter } from 'next/router'
import React from 'react'

import { QUESTION_TYPES_SHORT } from 'shared-components/src/constants'

interface SessionProps {
  sessionName: string
  sessionList: SessionType[]
}

const defaultProps = {}

function Session({
  sessionName,
  sessionList,
}: SessionProps): React.ReactElement {
  const [startSession] = useMutation(StartSessionDocument)
  const router = useRouter()

  return (
    <div>
      <H2 className="mb-2">{sessionName}</H2>
      <div className="mb-8">
        {sessionList.map((session: SessionType) => (
          <div key={session.id} className="mb-4">
            <div className="flex flex-row items-end border-b border-solid border-uzh-grey-40">
              <H4 className="flex-1">{session.displayName}</H4>
              {(session.status === 'RUNNING' ||
                session.status === 'COMPLETED') && (
                <Link href={`/sessions/${session.id}/evaluation`} passHref>
                  <a className="mr-4 text-sm hover:text-uzh-red-100">
                    Zur Evaluation
                  </a>
                </Link>
              )}
              <div className="text-sm">
                <FontAwesomeIcon icon={faCalendarDays} className="mr-1" />
                {dayjs(session.createdAt).format('YYYY-MM-DD HH:mm')}
              </div>
            </div>
            <div className="flex flex-row">
              <div className="flex flex-row flex-1">
                {session.blocks?.map((block: SessionBlock, index: number) => (
                  <div className="max-h-40" key={block.id}>
                    <div className="mb-1 ml-1 text-sm">Block {index}</div>
                    {block.instances.map((instance: QuestionInstance) => (
                      <div
                        key={instance.id}
                        className="flex flex-row p-0.5 border border-solid border-uzh-grey-60 rounded-md m-0.5 text-sm w-40"
                      >
                        <div>
                          {instance.questionData.name} (
                          {QUESTION_TYPES_SHORT[instance.questionData.type]})
                        </div>
                      </div>
                    ))}
                  </div>
                ))}
              </div>
              {session.status !== 'RUNNING' && session.status !== 'COMPLETED' && (
                <Button
                  className="px-2 mt-1 text-sm h-9 border-uzh-grey-80"
                  onClick={async () => {
                    await startSession({ variables: { id: session.id } })
                    router.push(`sessions/${session.id}/cockpit`)
                  }}
                >
                  <Button.Icon>
                    <FontAwesomeIcon icon={faPlay} className="mr-1" />
                  </Button.Icon>
                  <Button.Label>Start Session</Button.Label>
                </Button>
              )}
            </div>
          </div>
        ))}
      </div>
    </div>
  )
}

Session.defaultProps = defaultProps

export default Session
