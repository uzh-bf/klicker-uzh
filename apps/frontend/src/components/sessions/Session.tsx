import React, { useState } from 'react'
import dayjs from 'dayjs'
import Link from 'next/link'
import _get from 'lodash/get'
import { useMutation } from '@apollo/client'
import { Confirm, Button, Label, Icon, Message, Dropdown } from 'semantic-ui-react'
import { defineMessages, FormattedMessage, useIntl } from 'react-intl'
import clsx from 'clsx'

import SessionListQuery from '../../graphql/queries/SessionListQuery.graphql'
import DeleteSessionsMutation from '../../graphql/mutations/DeleteSessionsMutation.graphql'
import QuestionBlock from '../questions/QuestionBlock'
import { SESSION_STATUS } from '../../constants'

const messages = defineMessages({
  deletionConfirmationCancel: {
    defaultMessage: 'Cancel',
    id: 'session.deletionConfirmation.cancel',
  },
  deletionConfirmationConfirm: {
    defaultMessage: 'Delete session!',
    id: 'session.deletionConfirmation.confirm',
  },
  deletionConfirmationContent: {
    defaultMessage: 'Are you sure you want to delete "{name}"? All results will be lost!',
    id: 'session.deletionConfirmation.content',
  },
  storageModeComplete: {
    defaultMessage: 'COMPLETE',
    id: 'session.storageMode.complete',
  },
  storageModeSecret: {
    defaultMessage: 'AGGREGATED',
    id: 'session.storageMode.secret',
  },
  authModePassword: {
    defaultMessage: 'PASS',
    id: 'session.authMode.password',
  },
  authModeAAI: {
    defaultMessage: 'AAI',
    id: 'session.authMode.aai',
  },
})

interface Props {
  blocks?: any[]
  button: {
    disabled: boolean
    hidden?: boolean
    icon: 'copy' | 'play'
    message: React.ReactNode
    onClick: any
  }
  createdAt: string
  id: string
  name: string
  status: string
  isParticipantAuthenticationEnabled?: boolean
  authenticationMode?: 'PASSWORD' | 'AAI' | 'NONE'
  storageMode?: string
  confusionTS?: any[]
  feedbacks?: any[]
}

const defaultProps = {
  blocks: [],
  isParticipantAuthenticationEnabled: false,
  authenticationMode: 'NONE',
  storageMode: '',
  confusionTS: [],
  feedbacks: [],
}

function Session({
  button,
  createdAt,
  id,
  name,
  blocks,
  status,
  isParticipantAuthenticationEnabled,
  authenticationMode,
  storageMode,
  confusionTS,
  feedbacks,
}: Props): React.ReactElement {
  const intl = useIntl()

  const [deletionConfirmation, setDeletionConfirmation] = useState(false)

  const [deleteSessions] = useMutation(DeleteSessionsMutation)

  const isFeedbackSession = blocks.length === 0

  return (
    <div className="flex flex-col flex-1 md:flex-row md:flex-wrap">
      <h2 className="text-xl md:flex-00half md:m-0 md:mb-[0.4rem] mb-[0.3rem] text-primary-strong">{name}</h2>

      <div className="flex flex-row mb-1 labels md:flex-00half md:m-0 md:mb-[0.4rem] justify-end">
        <Label
          className="md:!order-3 md:!w-[11rem] text-center"
          content={dayjs(createdAt).format('DD.MM.YY HH:mm')}
          icon="calendar"
          style={{ marginRight: '20px !important' }}
        />
        {isParticipantAuthenticationEnabled && [
          <Label
            className="md:order-2 min-w-[5.8rem] text-center"
            color="green"
            content="AUTH"
            icon="shield alternate"
            key="authMode"
          />,
          authenticationMode === 'PASSWORD' && (
            <Label
              className="md:order-1 min-w-[5.8rem] text-center"
              content={intl.formatMessage(messages.authModePassword)}
              icon="key"
            />
          ),
          authenticationMode === 'AAI' && (
            <Label
              className="md:order-1 min-w-[5.8rem] text-center"
              content={intl.formatMessage(messages.authModeAAI)}
              icon="key"
            />
          ),
          storageMode === 'SECRET' && (
            <Label
              className="md:order-1 min-w-[5.8rem] text-center"
              content={intl.formatMessage(messages.storageModeSecret)}
              icon="user secret"
            />
          ),
          storageMode === 'COMPLETE' && (
            <Label
              className="md:order-1 min-w-[5.8rem] text-center"
              content={intl.formatMessage(messages.storageModeComplete)}
              icon="archive"
            />
          ),
        ]}
      </div>

      <div className="flex flex-col flex-1 md:flex-row md:flex-wrap">
        {isFeedbackSession && (
          <div className="md:flex-1 md:m-0 md:mr-[0.4rem] mt-[0.3rem]">
            <Message info>
              <FormattedMessage
                defaultMessage="This feedback session does not contain any questions."
                id="runningSession.message.feedbackSession"
              />
            </Message>
          </div>
        )}

        {blocks.map(
          ({ id: blockId, instances, timeLimit }): React.ReactElement => (
            <div className="md:flex-1 md:m-0 md:mr-[0.4rem] mt-[0.3rem]" key={blockId}>
              <QuestionBlock
                noDetails
                questions={instances.map(({ id: instanceId, question, version, results }): any => ({
                  id: instanceId,
                  title: question.title,
                  totalParticipants: _get(results, 'totalParticipants') || 0,
                  type: question.type,
                  version,
                }))}
                timeLimit={timeLimit}
              />
            </div>
          )
        )}

        <div className="flex flex-col actionArea md:self-start md:w-[11rem]">
          <Dropdown button labeled className="icon left" icon="wrench" text="Options">
            <Dropdown.Menu>
              {status === SESSION_STATUS.CREATED && (
                <Link passHref href={{ pathname: '/questions', query: { editSessionId: id } }}>
                  <Dropdown.Item>
                    <Icon name="edit" />
                    <FormattedMessage defaultMessage="Modify Session" id="session.button.modify" />
                  </Dropdown.Item>
                </Link>
              )}

              <Link
                passHref
                href={{
                  pathname: '/questions',
                  query: { copy: true, editSessionId: id },
                }}
              >
                <Dropdown.Item>
                  <Icon name="copy" />
                  <FormattedMessage defaultMessage="Copy & Modify" id="session.button.copyAndModify" />
                </Dropdown.Item>
              </Link>

              {status !== SESSION_STATUS.RUNNING && (
                <>
                  <Dropdown.Divider />

                  <>
                    <Dropdown.Item onClick={(): void => setDeletionConfirmation(true)}>
                      <Icon name="trash" />
                      <FormattedMessage defaultMessage="Delete Session" id="session.button.delete" />
                    </Dropdown.Item>
                    <Confirm
                      cancelButton={intl.formatMessage(messages.deletionConfirmationCancel)}
                      confirmButton={intl.formatMessage(messages.deletionConfirmationConfirm)}
                      content={intl.formatMessage(messages.deletionConfirmationContent, { name })}
                      open={deletionConfirmation}
                      onCancel={(): void => setDeletionConfirmation(false)}
                      onConfirm={async (): Promise<void> => {
                        try {
                          await deleteSessions({
                            optimisticResponse: {
                              __typename: 'Mutation',
                              deleteSessions: 'DELETION_SUCCESSFUL',
                            },
                            update: (cache, { data }): void => {
                              if (data.deleteSessions !== 'DELETION_SUCCESSFUL') {
                                return
                              }

                              const { sessions } = cache.readQuery({ query: SessionListQuery })
                              cache.writeQuery({
                                data: {
                                  sessions: sessions.filter((session): boolean => session.id !== id),
                                },
                                query: SessionListQuery,
                              })

                              setDeletionConfirmation(false)
                            },
                            variables: {
                              ids: [id],
                            },
                          })
                        } catch (e) {
                          console.error(e)
                        }

                        setDeletionConfirmation(false)
                      }}
                    />
                  </>
                </>
              )}
            </Dropdown.Menu>
          </Dropdown>

          {status !== SESSION_STATUS.CREATED && (
            <a
              className={clsx(
                isFeedbackSession &&
                  confusionTS.length === 0 &&
                  feedbacks.length === 0 &&
                  'pointer-events-none cursor-default'
              )}
              href={`/sessions/evaluation/${id}`}
              rel="noopener noreferrer"
              target="_blank"
            >
              <Button
                icon
                primary
                disabled={isFeedbackSession && confusionTS.length === 0 && feedbacks.length === 0}
                labelPosition="left"
              >
                <Icon name="external" />
                <FormattedMessage defaultMessage="Evaluation" id="session.button.evaluation" />
              </Button>
            </a>
          )}

          {button && !button.hidden && (
            <Button icon primary disabled={button.disabled} labelPosition="left" onClick={button.onClick}>
              <Icon name={button.icon} />
              {button.message}
            </Button>
          )}
        </div>
      </div>

      <style jsx>{`
        @import 'src/theme';

        .labels {
          :global(.label) {
            margin-right: 0.2rem !important;
          }
        }

        .actionArea {
          :global(.button),
          a :global(.button) {
            text-align: center;
            margin-right: 0 !important;
            width: 100%;
            margin-top: 0.3rem !important;
          }
        }

        @include desktop-tablet-only {
          .labels {
            :global(.label) {
              margin-right: 0 !important;
              margin-left: 0.4rem !important;
            }
          }

          .actionArea {
            :global(.button),
            a :global(.button) {
              margin-top: 0 !important;
              margin-bottom: 0.4rem !important;
            }

            :global(.button:last-child),
            a:last-child :global(.button) {
              margin-bottom: 0 !important;
            }
          }
        }
      `}</style>
    </div>
  )
}

Session.defaultProps = defaultProps

export default Session
