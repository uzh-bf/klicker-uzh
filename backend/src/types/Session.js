module.exports = `
  enum Session_Status {
    CREATED
    RUNNING
    PAUSED
    COMPLETED
  }

  enum Session_QuestionBlockStatus {
    PLANNED
    ACTIVE
    EXECUTED
  }

  enum Session_StorageMode {
    SECRET
    COMPLETE
  }

  enum Session_AuthenticationMode {
    PASSWORD
    AAI
    NONE
  }

  type Session_Public {
    id: ID!

    status: Session_Status!
    timeLimit: Int
    expiresAt: DateTime
    execution: Int
    isFeedbackOnlySession: Boolean

    settings: Session_Settings!

    activeInstances: [Question_Public]!
    feedbacks: [Session_Feedback!]
  }

  type Session_Public_Update {
    id: ID!

    status: Session_Status!
    timeLimit: Int
    expiresAt: DateTime
    execution: Int
    isFeedbackOnlySession: Boolean

    settings: Session_Settings!

    activeInstances: [Question_Public]!
  }

  input SessionInput {
    name: String!
    blocks: [Session_QuestionBlockInput!]!
    participants: [Session_ParticipantInput!]
    authenticationMode: Session_AuthenticationMode
    storageMode: Session_StorageMode
  }
  input SessionModifyInput {
    name: String
    blocks: [Session_QuestionBlockInput!]
    participants: [Session_ParticipantInput!]
    authenticationMode: Session_AuthenticationMode
    storageMode: Session_StorageMode
  }
  type Session {
    id: ID!
    name: String!
    activeBlock: Int!
    activeStep: Int!
    execution: Int

    status: Session_Status!
    settings: Session_Settings!
    user: User!

    blocks: [Session_QuestionBlock!]!
    confusionTS: [Session_ConfusionTimestep!]!
    feedbacks: [Session_Feedback!]!
    participants: [Session_Participant]!

    createdAt: DateTime!
    updatedAt: DateTime!
    startedAt: DateTime!
    finishedAt: DateTime!
  }
  type Session_PublicEvaluation {
    id: ID!
    status: Session_Status!

    blocks: [Session_QuestionBlock_Public!]!
  }

  input Session_SettingsInput {
    isConfusionBarometerActive: Boolean
    isEvaluationPublic: Boolean
    isFeedbackChannelActive: Boolean
    isFeedbackChannelPublic: Boolean
  }

  type Session_Settings {
    isParticipantAuthenticationEnabled: Boolean!
    isConfusionBarometerActive: Boolean!
    isEvaluationPublic: Boolean!
    isFeedbackChannelActive: Boolean!
    isFeedbackChannelPublic: Boolean!
    authenticationMode: Session_AuthenticationMode!
    storageMode: Session_StorageMode!
  }

  input Session_ParticipantInput {
    username: String!
    isAAI: Boolean
  }

  type Session_Participant {
    id: ID!

    username: String!
    password: String
  }

  input Session_QuestionBlockQuestionInput {
    question: ID!
    version: Int!
  }
  input Session_QuestionBlockInput {
    timeLimit: Int
    randomSelection: Int
    questions: [Session_QuestionBlockQuestionInput!]!
  }
  input Session_QuestionBlockModifyInput {
    timeLimit: Int
    randomSelection: Int
  }
  type Session_QuestionBlock {
    id: ID!

    status: Session_QuestionBlockStatus!
    timeLimit: Int
    randomSelection: Int
    expiresAt: DateTime

    instances: [QuestionInstance!]!
  }
  type Session_QuestionBlock_Public {
    id: ID!
    status: Session_QuestionBlockStatus!

    instances: [QuestionInstance_Public!]!
  }

  type Session_ConfusionTimestep {
    id: ID!
    difficulty: Int!
    speed: Int!

    createdAt: DateTime!
  }

  type Session_Feedback {
    id: ID!
    content: String!
    votes: Int!
    resolved: Boolean!
    pinned: Boolean!
    published: Boolean!

    responses: [Session_FeedbackResponse!]!

    createdAt: DateTime!
    resolvedAt: DateTime
  }

  type Session_Feedback_Public {
    id: ID!
    content: String!
    resolved: Boolean!

    responses: [Session_FeedbackResponse_Public!]!

    createdAt: DateTime!
    resolvedAt: DateTime
  }

  type Session_Feedback_ResolvedStateChange {
    feedbackId: ID!
    resolvedState: Boolean!
    resolvedAt: DateTime
  }

  type Session_FeedbackResponse {
    id: ID!

    content: String!
    positiveReactions: Int!
    negativeReactions: Int!

    createdAt: DateTime!
  }

  type Session_FeedbackResponse_Public {
    id: ID!

    content: String!

    createdAt: DateTime!
  }

  type Session_FeedbackResponse_Added {
    feedbackId: ID!
    id: ID!

    content: String!

    createdAt: DateTime!
  }

  type Session_FeedbackResponse_Deleted {
    feedbackId: ID!
    id: ID!
  }

  type Session_Update {
    id: ID!
    activeStep: Int!
    activeBlock: Int!

    blocks: [SessionUpdate_Block!]!
  }

  type SessionUpdate_Block {
    id: ID!
    status: Session_QuestionBlockStatus!
    timeLimit: Int
    randomSelection: Int
    expiresAt: DateTime
  }
`
