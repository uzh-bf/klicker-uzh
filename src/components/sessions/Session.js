// @flow

import * as React from 'react'
import moment from 'moment'
import { Button, Icon } from 'semantic-ui-react'
import { FormattedMessage } from 'react-intl'

import QuestionBlock from '../questions/QuestionBlock'

type Props = {
  blocks: Array<{
    id: string,
    questions: Array<{
      id: string,
      questionDefinition: {
        title: string,
        type: string,
      },
    }>,
    showSolutions: boolean,
    timeLimit: number,
  }>,
  createdAt: string,
  id: string,
  name: string,
  status: 'CREATED' | 'RUNNING' | 'COMPLETED',
}

const Session = ({ createdAt, name, blocks, id, status }: Props) => {
  const statusCases = {
    COMPLETED: {
      disabled: false,
      icon: 'copy',
      message: <FormattedMessage id="session.button.completed.content" defaultMessage="Copy" />,
    },
    CREATED: {
      disabled: false,
      icon: 'play',
      message: <FormattedMessage id="session.button.created.content" defaultMessage="Start" />,
    },
    RUNNING: {
      disabled: true,
      icon: 'play',
      message: <FormattedMessage id="session.button.running.content" defaultMessage="Running" />,
    },
  }
  const buttonStatus = statusCases[status]

  return (
    <div className="session">
      <h2 className="title">
        {id.slice(0, 7)} - {name}
      </h2>
      <div className="date">
        <FormattedMessage id="sessionHistory.string.createdOn" defaultMessage="Created on" />{' '}
        {moment(createdAt).format('DD.MM.YYYY HH:MM:SS')}
      </div>

      <div className="details">
        {blocks.map(block => (
          <div className="block">
            <QuestionBlock
              key={block.id}
              questions={block.questions.map(question => ({
                id: question.id,
                title: question.questionDefinition.title,
                type: question.questionDefinition.type,
              }))}
              showSolutions={block.showSolutions}
              timeLimit={block.timeLimit}
            />
          </div>
        ))}
        <div className="actionArea">
          <Button icon disabled={buttonStatus.disabled} labelPosition="left">
            <Icon name={buttonStatus.icon} />
            {buttonStatus.message}
          </Button>
        </div>
      </div>

      <style jsx>{`
        .session,
        .details {
          display: flex;
          flex-direction: column;
        }
        .title,
        .date {
          margin: auto;
          margin-bottom: 0.5rem;
        }
        .block {
          margin: 0 0.5rem;
          margin-bottom: 0.5rem;
        }
        .actionArea {
          margin: auto;
        }

        @media all and (min-width: 768px) {
          .session,
          .details {
            flex-flow: row wrap;
          }
          .title,
          .date {
            flex: 0 0 50%;
            margin: 0;
          }
          .title {
            font-size: 1.2rem;
            margin-bottom: 0.5rem;
          }
          .date {
            align-self: center;
            text-align: right;
          }
          .details {
            border: 1px solid lightgrey;
          }
          .block {
            flex: 1;
            margin: 0.3rem;
          }
          .actionArea {
            align-self: flex-end;
            margin: 0 0 0.3rem 0.3rem;
          }
        }
      `}</style>
    </div>
  )
}

export default Session
