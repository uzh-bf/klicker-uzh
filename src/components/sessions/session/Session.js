import React from 'react'
import PropTypes from 'prop-types'
import moment from 'moment'
import { FormattedMessage } from 'react-intl'
import { Button, Icon } from 'semantic-ui-react'

import QuestionBlock from './QuestionBlock'
import withCSS from '../../../lib/withCSS'

const Session = ({ createdAt, name, blocks, id, status }) => {
  const statusCases = {
    completed: { disabled: false, icon: 'copy', message: <FormattedMessage id="session.button.completed.content" defaultMessage="Copy" /> },
    created: { disabled: false, icon: 'play', message: <FormattedMessage id="session.button.created.content" defaultMessage="Start" /> },
    running: { disabled: true, icon: 'play', message: <FormattedMessage id="session.button.running.content" defaultMessage="Running" /> },
  }
  const buttonStatus = statusCases[status]

  return (
    <div className="session">
      <h2 className="title">
        {id.slice(0, 7)} - {name}
      </h2>
      <div className="date">
        <FormattedMessage id="sessionHistory.string.createdOn" defaultMessage="Created on" />{' '}
        {moment(createdAt).format('DD.MM.YYYY HH:MM:SS')}
      </div>

      <div className="details">
        {blocks.map(block =>
          (<div className="block">
            <QuestionBlock
              key={block.id}
              questions={block.questions.map(question => ({
                id: question.id,
                title: question.questionDefinition.title,
                type: question.questionDefinition.type,
              }))}
              showSolutions={block.showSolutions}
              timeLimit={block.timeLimit}
            />
          </div>),
        )}
        <div className="actionArea">
          <Button icon disabled={buttonStatus.disabled} labelPosition="left">
            <Icon name={buttonStatus.icon} />
            {buttonStatus.message}
          </Button>
        </div>
      </div>

      <style jsx>{`
        .session, .details {
          display: flex;
          flex-direction: column;
        }
        .title,
        .date {
          margin: auto;
          margin-bottom: 0.5rem;
        }
        .block {
          margin: 0 0.5rem;
          margin-bottom: 0.5rem;
        }
        .actionArea {
          margin: auto;
        }

        @media all and (min-width: 768px) {
          .session, .details {
            flex-flow: row wrap;
          }
          .title,
          .date {
            flex: 0 0 50%;
            margin: 0;
          }
          .title {
            font-size: 1.2rem;
            margin-bottom: 0.5rem;
          }
          .date {
            align-self: center;
            text-align: right;
          }
          .details {
            border: 1px solid lightgrey;
          }
          .block {
            flex: 1;
            margin: 0.3rem;
          }
          .actionArea {
            align-self: flex-end;
            margin: 0 0 0.3rem 0.3rem;
          }
        }
      `}</style>
    </div>
  )
}

Session.propTypes = {
  blocks: PropTypes.shape({
    questions: PropTypes.shape({
      id: PropTypes.string.isRequired,
      questionDefinition: PropTypes.shape({
        title: PropTypes.string.isRequired,
        type: PropTypes.string.isRequired,
      }).isRequired,
      showSolutions: PropTypes.bool.isRequired,
      timeLimit: PropTypes.number.isRequired,
    }),
  }).isRequired,
  createdAt: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  status: PropTypes.string.isRequired,
}

Session.defaultProps = {
  status: 'created',
}

export default withCSS(Session, ['segment'])
