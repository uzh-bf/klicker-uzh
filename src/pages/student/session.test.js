/* eslint-env jest */

import * as React from 'react'
import renderer from 'react-test-renderer'

import Session from './session'

// HACK: workaround for https://github.com/Semantic-Org/Semantic-UI-React/issues/1702
jest.mock('react-dom', () => ({
  findDOMNode: jest.fn(() => {}),
}))

describe('Snapshot-Testing', () => {
  it.skip('Works', () => {
    const component = renderer.create(<Session />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
