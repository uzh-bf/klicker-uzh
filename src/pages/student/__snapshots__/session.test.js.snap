// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot-Testing Works 1`] = `
<div
  className="studentLayout"
  data-jsx={3095557183}
>
  <div
    className="header"
    data-jsx={3095557183}
  >
    <button
      className="ui basic icon button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      <i
        aria-hidden="true"
        className="content icon"
      />
    </button>
    <h1
      data-jsx={3095557183}
    >
      Active Question
    </h1>
  </div>
  <div
    className="content"
    data-jsx={3095557183}
  >
    <div
      className="sidebar"
      data-jsx={3743778143}
    >
      <div
        className="pushable"
      >
        <div
          className="ui vertical labeled icon ui overlay left wide sidebar sidebarMenu menu"
        >
          <a
            className="active item"
            onClick={[Function]}
          >
            <span>
              Active Question
            </span>
          </a>
          <a
            className="item"
            onClick={[Function]}
          >
            <span>
              Feedback-Channel
            </span>
          </a>
        </div>
        <div
          className="pusher"
        >
          <div
            className="student"
            data-jsx={3709880010}
          >
            <div
              className="questionArea active"
              data-jsx={3709880010}
            >
              <div
                className="collapser"
                data-jsx={3709880010}
              >
                <div
                  className="collapser"
                  data-jsx={1840183473}
                >
                  <div
                    className="content collapsed"
                    data-jsx={1840183473}
                  >
                    <p
                      data-jsx={3709880010}
                    >
                      hello this is a very short question that is getting longer and longer as we speak. it is in fact very very long. the end is even hidden at the beginning.
                    </p>
                    <p
                      data-jsx={3709880010}
                    >
                      wow, is this a long question. i could never have imagined seeing such a question.
                    </p>
                    <p
                      data-jsx={3709880010}
                    >
                      hello this is a very short question that is getting longer and longer as we speak. it is in fact very very long. the end is even hidden at the beginning.
                    </p>
                    <p
                      data-jsx={3709880010}
                    >
                      wow, is this a long question. i could never have imagined seeing such a question.
                    </p>
                  </div>
                  <div
                    className="collapse"
                    data-jsx={1840183473}
                  >
                    <i
                      aria-hidden="true"
                      className="angle double down icon"
                      onClick={[Function]}
                    />
                  </div>
                </div>
              </div>
              <div
                className="options"
                data-jsx={3709880010}
              >
                <div
                  className="options"
                  data-jsx={964719302}
                >
                  <div
                    className="option"
                    data-jsx={964719302}
                  >
                    <button
                      className="ui basic fluid button"
                      disabled={undefined}
                      onClick={[Function]}
                      role="button"
                      tabIndex={undefined}
                    >
                      answer1
                    </button>
                  </div>
                  <div
                    className="option"
                    data-jsx={964719302}
                  >
                    <button
                      className="ui basic fluid button"
                      disabled={undefined}
                      onClick={[Function]}
                      role="button"
                      tabIndex={undefined}
                    >
                      antwort 2
                    </button>
                  </div>
                  <div
                    className="option"
                    data-jsx={964719302}
                  >
                    <button
                      className="ui basic fluid button"
                      disabled={undefined}
                      onClick={[Function]}
                      role="button"
                      tabIndex={undefined}
                    >
                      option 3
                    </button>
                  </div>
                  <div
                    className="option"
                    data-jsx={964719302}
                  >
                    <button
                      className="ui basic fluid button"
                      disabled={undefined}
                      onClick={[Function]}
                      role="button"
                      tabIndex={undefined}
                    >
                      tschege
                    </button>
                  </div>
                </div>
              </div>
              <div
                className="actionButton"
                data-jsx={3709880010}
              >
                <button
                  className="ui primary button submitButton"
                  disabled={undefined}
                  onClick={[Function]}
                  role="button"
                  tabIndex={undefined}
                >
                  <span>
                    Send
                  </span>
                </button>
              </div>
            </div>
            <div
              className="feedbackArea"
              data-jsx={3709880010}
            >
              <div
                className="confusion"
                data-jsx={3709880010}
              >
                <div
                  className="confusionSlider"
                  data-jsx={219419674}
                >
                  <div
                    className="title"
                    data-jsx={219419674}
                  >
                    <h2
                      data-jsx={3709880010}
                    >
                      <span>
                        Speed
                      </span>
                    </h2>
                  </div>
                  <div
                    className="slider"
                    data-jsx={219419674}
                  >
                    <div
                      className="rangeslider rangeslider-horizontal"
                      onMouseDown={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchStart={[Function]}
                    >
                      <div
                        className="rangeslider__fill"
                        style={
                          Object {
                            "width": "0px",
                          }
                        }
                      />
                      <div
                        className="rangeslider__handle"
                        onMouseDown={[Function]}
                        onTouchEnd={[Function]}
                        onTouchMove={[Function]}
                        style={
                          Object {
                            "left": "0px",
                          }
                        }
                      >
                        <div
                          className="rangeslider__tooltip"
                        >
                          <span />
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  className="confusionSlider"
                  data-jsx={219419674}
                >
                  <div
                    className="title"
                    data-jsx={219419674}
                  >
                    <h2
                      data-jsx={3709880010}
                    >
                      <span>
                        Difficulty
                      </span>
                    </h2>
                  </div>
                  <div
                    className="slider"
                    data-jsx={219419674}
                  >
                    <div
                      className="rangeslider rangeslider-horizontal"
                      onMouseDown={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchStart={[Function]}
                    >
                      <div
                        className="rangeslider__fill"
                        style={
                          Object {
                            "width": "0px",
                          }
                        }
                      />
                      <div
                        className="rangeslider__handle"
                        onMouseDown={[Function]}
                        onTouchEnd={[Function]}
                        onTouchMove={[Function]}
                        style={
                          Object {
                            "left": "0px",
                          }
                        }
                      >
                        <div
                          className="rangeslider__tooltip"
                        >
                          <span />
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="feedbacks"
                data-jsx={3709880010}
              >
                <div
                  className="feedback"
                  data-jsx={3709880010}
                >
                  <div
                    className="feedback"
                    data-jsx={3449085289}
                  >
                    <div
                      className="content"
                      data-jsx={3449085289}
                    >
                      hello world
                    </div>
                    <div
                      className="votes"
                      data-jsx={3449085289}
                    >
                      +
                      100
                    </div>
                  </div>
                </div>
                <div
                  className="feedback"
                  data-jsx={3709880010}
                >
                  <div
                    className="feedback"
                    data-jsx={3449085289}
                  >
                    <div
                      className="content"
                      data-jsx={3449085289}
                    >
                      blablbla
                    </div>
                    <div
                      className="votes"
                      data-jsx={3449085289}
                    >
                      +
                      50
                    </div>
                  </div>
                </div>
                <div
                  className="feedback"
                  data-jsx={3709880010}
                >
                  <div
                    className="feedback"
                    data-jsx={3449085289}
                  >
                    <div
                      className="content"
                      data-jsx={3449085289}
                    >
                      hello
                    </div>
                    <div
                      className="votes"
                      data-jsx={3449085289}
                    >
                      +
                      25
                    </div>
                  </div>
                </div>
                <div
                  className="feedback"
                  data-jsx={3709880010}
                >
                  <div
                    className="feedback"
                    data-jsx={3449085289}
                  >
                    <div
                      className="content"
                      data-jsx={3449085289}
                    >
                      hooi
                    </div>
                    <div
                      className="votes"
                      data-jsx={3449085289}
                    >
                      +
                      10
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="actionButton"
                data-jsx={3709880010}
              >
                <button
                  className="ui large circular icon primary button"
                  disabled={undefined}
                  onClick={[Function]}
                  role="button"
                  tabIndex={undefined}
                >
                  <i
                    aria-hidden="true"
                    className="plus icon"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
